controller/openid_configuration.go

```diff
@@ -0,0 +1,62 @@
+package controller
+
+import (
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/client"
+	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/goadesign/goa"
+)
+
+// OpenidConfigurationController implements the openid_configuration resource.
+type OpenidConfigurationController struct {
+	*goa.Controller
+}
+
+// NewOpenidConfigurationController creates a openid_configuration controller.
+func NewOpenidConfigurationController(service *goa.Service) *OpenidConfigurationController {
+	return &OpenidConfigurationController{Controller: service.NewController("OpenidConfigurationController")}
+}
+
+// Show runs the show action.
+func (c *OpenidConfigurationController) Show(ctx *app.ShowOpenidConfigurationContext) error {
+
+	issuer := rest.AbsoluteURL(ctx.RequestData, "")
+	authorizationEndpoint := rest.AbsoluteURL(ctx.RequestData, client.AuthorizeAuthorizePath())
+	tokenEndpoint := rest.AbsoluteURL(ctx.RequestData, client.ExchangeTokenPath())
+	logoutEndpoint := rest.AbsoluteURL(ctx.RequestData, client.LogoutLogoutPath())
+	// jwksURI := c.configuration.GetKeycloakEndpointCerts()//
+	jwksURI := rest.AbsoluteURL(ctx.RequestData, client.KeysTokenPath())
+
+	authOpenIDConfiguration := &app.OpenIDConfiguration{
+		// REQUIRED properties
+		Issuer:                 &issuer,
+		AuthorizationEndpoint:  &authorizationEndpoint,
+		TokenEndpoint:          &tokenEndpoint,
+		EndSessionEndpoint:     &logoutEndpoint,
+		ResponseTypesSupported: []string{"code"},
+		JwksURI:                &jwksURI,
+
+		// subject_types_supported REQUIRED. JSON array containing a list of the Subject Identifier types that this OP supports. Valid types include pairwise and public.
+		// A Subject Identifier is a locally unique and never reassigned identifier within the Issuer for the End-User, which is intended to be consumed by the Client.
+		// http://openid.net/specs/openid-connect-core-1_0.html#SubjectIDTypes
+		// subject ==  identity
+		SubjectTypesSupported: []string{"public"},
+		// id_token_signing_alg_values_supported REQUIRED. JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP for the ID Token to encode the Claims in a JWT [JWT].
+		// The algorithm RS256 MUST be included. The value none MAY be supported, but MUST NOT be used unless the Response Type used returns no ID Token from the Authorization Endpoint (such as when using the Authorization Code Flow).
+		IDTokenSigningAlgValuesSupported: []string{"RS256"},
+
+		// RECOMMENDED properties
+		// userinfo_endpoint : We don't have a OpenID compliant userinfo_endpoint
+		ScopesSupported: []string{"openid", "offline_access"},
+		ClaimsSupported: []string{"email", "full_name", "email", "image_url", "bio", "url", "company", "cluster", "email_verified", "email_private", "feature_level"},
```

Comment:
There are two "email"
We do not include image, bio, url, cluster, email_private, and feature_level to the token.
Claims we DO include:
```
    "sub",
    "iss",
    "auth_time",
    "name",
    "given_name",
    "family_name",
    "preferred_username",
    "email"
```

Reply:
yeah, I was a little confused about this one. I picked columns from the users table.